include/InfoLogger/InfoLoggerFMQ.hxx

```diff
@@ -91,8 +91,8 @@ void setFMQLogsToInfoLogger(AliceO2::InfoLogger::InfoLogger* logPtr = nullptr)
         severity,
         level,
         AliceO2::InfoLogger::InfoLogger::undefinedMessageOption.errorCode,
-        metadata.file.c_str(),
-        atoi(metadata.line.c_str())
+        std::string(metadata.file.data(), metadata.file.size()).c_str(),
```

Comment:
Does InfoLoggerMessageOption copy the string?

Reply:
No copy, InfoLoggerMessageOption is a POD with a char*. It's meant to be resolved at compile time whenever possible.

Reply:
I replaced the char* in InfoLoggerMessageOption with a string_view, plus subsequent changes

Reply:
@sy-c thanks for taking care of it.

simply creating the string here https://github.com/AliceO2Group/InfoLogger/commit/940fefa6bc5a7fb2ea3ab5ed2f07835a28c7ad64#diff-36584f894efde34aee0e3aedfd701712f9a7767d6bd748c11dc604bfd937d9acL94 was also my first thought. But @ktf  reminded me that `std::string(metadata.file).c_str()` will go out of scope and the const char* of `AliceO2::InfoLogger::InfoLogger::InfoLoggerMessageOption opt` will point to something undefined. Or am I missing something different?

Reply:
You are right, sorry for this. Please use v2.8.1. https://github.com/AliceO2Group/InfoLogger/pull/97

Valgrind did not complained in my tests, I guess the compiler optimized it as a singleton in the log() call - which is how it is usually written: InfoLoggerMessageOption directly defined in the log() call without defining a variable. Here with the IF and two log() we indeed need the intermediate step.

Reply:
>  I guess the compiler optimized it as a singleton in the log() call - which is how it is usually written: InfoLoggerMessageOption directly set in the log() call without defining a variable.

I see, thanks.